app/services/audit/directory_checksum_validator.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+module Audit
+  # Service for validating Moab checksums in a directory. This class's primary collaborator
+  # is Audit::ChecksumValidator, which depends on Audit::DirectoryChecksumValidator to
+  # validate manifest inventories on disk and update the passed-in `results` object. It is
+  # also invoked by the `prescat:audit:validate_uncataloged` rake task which expects the
+  # class to emit validation results and errors to STDOUT.
```

Comment:
Try to be clear about the intended usage of this class and why it behaves differently sometimes.